nmpolicy/internal/resolver/path.go

```diff
@@ -32,7 +32,7 @@ func applyFuncOnPath(inputState interface{},
 	path []ast.Node,
 	expectedNode ast.Node,
 	funcToApply func(map[string]interface{}, string, ast.Node) (interface{}, error),
-	shouldFilterSlice bool) (interface{}, error) {
+	shouldFilterSlice bool, shouldFilterMap bool) (interface{}, error) {
```

Comment:
IIUC, this change is for readability only? Right?
The old code also return all the matching slices as is (without any map/slice filtering). And all the maps till the matching slice were filtered.
If so, it is not related to this commit/PR. 

EDIT: unless you accept my comment about not filtering at all in case of a replace. Than I understand why the change is relevant here.

Reply:
I am chacing it to not filter on replace so we need it.